### PR TITLE
Webpack bundling

### PR DESCRIPTION
**Pull request by edgarmueller**
Removes the need for grunt and bower and instead uses webpack and npm modules to create a bundle for easy client consumption. Code structure also has been refined slightly: framework code is now under src/. Furthermore module syntax has been introduced and obsolete files have been removed. Some minor things still need to be fixed, among them re-enabling protractor tests and bootstrap workflow configuration.

### CODE REVIEW

**Comment by edgarmueller** [comment]
@eneufeld Sorry, this is quite a monster..

File: src/components/renderers/controls/integer/integer-renderer.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        return control;
+    }
+
+    isApplicable(uiElement: IUISchemaElement, subSchema: SchemaElement, schemaPath: string):boolean {
+        return uiElement.type == 'Control' && subSchema !== undefined && subSchema.type == 'integer';
+    }
+}
+
+export default angular
+    .module('jsonforms.renderers.controls.integer', [])
```
Comment (eneufeld) [inline src/components/renderers/controls/integer/integer-renderer.ts]: i think a `jsonforms.renderers.controls`dependency is missing

File: src/components/generators/schema/schema-service.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (eneufeld) [inline src/components/generators/schema/schema-service.ts]: why don't register in jsonforms-schemagenerator just like you do it for renderer and thus remove the schema/uischema folders which would then contain exactly one class?

File: src/components/generators/uischema/uischema-service.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import {UISchemaGenerator} from "./jsonforms-uischemagenerator";
```
Comment (eneufeld) [inline src/components/generators/uischema/uischema-service.ts]: why not register in jsonforms-uischemagenerator, also see jsonforms-schemagenerator

File: src/components/renderers/layouts/masterdetail/masterdetail-renderer.ts [hunk 4]
```diff
@@ -38,6 +42,9 @@ class MasterDetailRenderer implements JSONForms.IRenderer {
     }
 }
 
-angular.module('jsonforms.renderers.layouts.masterdetail').run(['RenderService', (RenderService) => {
-    RenderService.register(new MasterDetailRenderer());
-}]);
+export default angular
```
Comment (eneufeld) [inline src/components/renderers/layouts/masterdetail/masterdetail-renderer.ts]: missing dependency to `jsonforms.renderers.layouts` in module registration

**Comment by eneufeld** [comment]
Looks good, thank you for the refactoring. 
I think there are some minor issues left

File: src/components/renderers/layouts/masterdetail/masterdetail-renderer.ts (hunk 4 above)
Comment (edgarmueller) [inline src/components/renderers/layouts/masterdetail/masterdetail-renderer.ts]: Registering the module happens in `masterdetail-renderer.ts`, so I think this ok.

**Comment by edgarmueller** [comment]
@eneufeld  Thanks for the feedback, I addressed your changes in the lastest commit.

**Comment by eneufeld** [comment]
Thank you for the huge refactoring

**Pull request merged**